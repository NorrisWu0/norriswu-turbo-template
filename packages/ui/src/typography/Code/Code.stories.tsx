import { ComponentMeta, ComponentStory } from "@storybook/react";

import { Code } from "./Code";

export default {
  title: "Typography / Code",
  component: Code,
  args: {
    children: `console.log("hello world")`,
  },
} as ComponentMeta<typeof Code>;

const Template: ComponentStory<typeof Code> = (args) => <Code {...args} />;

export const Default = Template.bind({});
