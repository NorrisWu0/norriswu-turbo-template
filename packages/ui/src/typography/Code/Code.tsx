import { Text } from "@chakra-ui/react";

type CodeProps = {
  children: string;
};
export function Code({ children }: CodeProps) {
  return (
    <Text as="code" fontWeight="700" fontFamily="var(--font-mono)">
      {children}
    </Text>
  );
}
