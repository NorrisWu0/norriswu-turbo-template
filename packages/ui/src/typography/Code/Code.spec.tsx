import { render } from "@testing-library/react";

import { Code } from "./Code";

describe("Code", () => {
  it("should render", () => {
    const { container } = render(<Code>Code</Code>);
    expect(container).toMatchSnapshot();
  });
});
