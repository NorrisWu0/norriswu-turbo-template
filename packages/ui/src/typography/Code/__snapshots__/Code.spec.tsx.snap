// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code should render 1`] = `
<div>
  <code
    class="chakra-text css-cjo7mn"
  >
    Code
  </code>
</div>
`;
